### PR TITLE
[netif] add multicast address support

### PR DESCRIPTION
**Pull request by gjc13**

### CODE REVIEW

**Review by bukepo — APPROVED** [review]
LGTM:+1:

File: src/core/netif.cpp [hunk 1]
```diff
@@ -234,18 +255,17 @@ static void processAddress(const otIp6Address *aAddress, uint8_t aPrefixLength,
 {
     (void)aContext;
 
+    bool isMulticast = (aPrefixLength == kMulticastPrefixLength);
```
Comment (jwhui) [inline src/core/netif.cpp]: Why not check for `ff00::/8` prefix instead?

File: src/core/netif.cpp [hunk 2]
```diff
@@ -234,18 +254,23 @@ static void processAddress(const otIp6Address *aAddress, uint8_t aPrefixLength,
 {
     (void)aContext;
 
+    // All multicast address has prefix ff00::/8
+    bool isMulticast = (aAddress->mFields.m8[0] == 0xff);
+    
+    for (int i = 0; i < 16; i++) {
```
Comment (bukepo) [inline src/core/netif.cpp]: remove debug message?

**Review by bukepo — APPROVED** [review]
LGTM:+1:

File: src/core/netif.cpp [hunk 3]
```diff
@@ -147,17 +147,26 @@ struct netif *otrGetNetif(void)
...
     err_t   err   = ERR_OK;
 
     LOCK_TCPIP_CORE();
+
     if (IsLinkLocal(aAddress))
     {
         netif_ip6_addr_set(&sNetif, 0, reinterpret_cast<const ip6_addr_t *>(&aAddress));
         netif_ip6_addr_set_state(&sNetif, 0, IP6_ADDR_PREFERRED);
     }
+    else if (isMulticast)
```
Comment (jwhui) [inline src/core/netif.cpp]: nit: is there a reason to overload the existing `addAddress()` and `delAddress()` methods? why not have a separate `addUnicastAddress()` and `addMulticastAddress()` methods?

**Review by jwhui — APPROVED** [review]

**Pull request merged**